test/scripts/e2e_subs/app-inner-calls-csp.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+
+import os
+import sys
+from goal import Goal
+import algosdk.logic as logic
+
+from datetime import datetime
+
+stamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+print(f"{os.path.basename(sys.argv[0])} start {stamp}")
+
+goal = Goal(sys.argv[1], autosend=True)
+
+joe = goal.new_account()
+
+txinfo1, err = goal.pay(goal.account, joe, amt=500_000)
+assert not err, err
+
+txinfo1, err = goal.keyreg(joe, nonpart=True)
+assert not err, err
+joeb = goal.balance(joe)
+
+# app1 calls the clear state program of app2 (after opting into it) which issues an
+# inner app call to app3. This verifies that both accessing a CSP through inner app
+# calls and issuing inner app calls from a CSP is possible.
+app1 = """
+#pragma version 6
+ txn ApplicationID
+ bz end
+
+ itxn_begin
+  int appl
+  itxn_field TypeEnum
+
+  txn Applications 1
+  dup
+  store 0
+  itxn_field ApplicationID
+
+  int OptIn
+  itxn_field OnCompletion
+
+  txn Applications 2
+  itxn_field Applications
+
+ itxn_next
+  int appl
+  itxn_field TypeEnum
+
+  load 0
+  itxn_field ApplicationID
+
+  txn Applications 2
+  itxn_field Applications
+
+  int ClearState
+  itxn_field OnCompletion
+
+ itxn_submit
+
+
+end:
+ int 1
+"""
+
+app2 = """
+#pragma version 6
+ txn ApplicationID
+ bz end
+
+ itxn_begin
+  int appl
+  itxn_field TypeEnum
+
+  txn Applications 1
+  itxn_field ApplicationID
+ itxn_submit
+
+
+ end:
+  int 1
+"""
+
+app3 = """
+#pragma version 6
+int 1
+"""
+
+goal.autosend = True
```

Comment:
@algoidurovic Is the highlighted statement needed for test execution?

I think I don't totally understand the implications of removing the configuration.  But, I see the test passes without it.

```suggestion

```

Reply:
It's meant to guarantee that the transaction building calls automatically send the transaction and wait for confirmation. This behavior is the default in some cases but I added it here explicitly to make my intentions clear and to prevent the test from failing if defaults change.

Reply:
@algoidurovic Gotcha - thanks for the explanation.  Resolving.